### PR TITLE
#287 Serializer fix for inclusion

### PR DESCRIPTION
**Pull request by mitch2na**

### CODE REVIEW

File: katharsis-core/src/main/java/io/katharsis/resource/ResourceIdentifier.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public int hashCode() {
 
 	@Override
 	public boolean equals(Object obj) {
-		if (obj == null || obj.getClass() != ResourceIdentifier.class)
+		if (obj == null || obj.getClass() != this.getClass())
```
Comment (chb0github) [inline katharsis-core/src/main/java/io/katharsis/resource/ResourceIdentifier.java]: An example of why not to implement your own `hashCode()` and `equals()`

File: katharsis-servlet/src/test/java/io/katharsis/servlet/resource/model/Locale.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package io.katharsis.servlet.resource.model;
+
+import io.katharsis.resource.annotations.JsonApiId;
+import io.katharsis.resource.annotations.JsonApiResource;
+
+/**
+ * Created by nickmitchell on 1/6/17.
+ */
+@JsonApiResource(type = "lang-locales")
+public class Locale {
```
Comment (chb0github) [inline katharsis-servlet/src/test/java/io/katharsis/servlet/resource/model/Locale.java]: Sorry, but I need to make my point a bit more on `@Lombok` -- this whole class could be reduced to 5 lines for identical functionality

File: katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/LocaleRepository.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.katharsis.queryParams.QueryParams;
+import io.katharsis.repository.ResourceRepository;
+import io.katharsis.servlet.resource.model.Locale;
+
+/**
+ * Created by nickmitchell on 1/6/17.
+ */
+public class LocaleRepository implements ResourceRepository<Locale, Long> {
```
Comment (chb0github) [inline katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/LocaleRepository.java]: I don't understand what the locale has to do with the serializer and why there is a repository

Comment (chb0github) [inline katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/LocaleRepository.java]: Also, there is already Abstract classes that implements all these methods and allows you to just override what you need. Please use that one if this class turns out to be necessary

File: katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/NodeCommentRepository.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.katharsis.queryParams.QueryParams;
+import io.katharsis.repository.ResourceRepository;
+import io.katharsis.servlet.resource.model.NodeComment;
+
+/**
+ * Created by nickmitchell on 1/6/17.
+ */
+public class NodeCommentRepository implements ResourceRepository<NodeComment, Long> {
```
Comment (chb0github) [inline katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/NodeCommentRepository.java]: See above

File: katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/NodeRepository.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.katharsis.queryParams.QueryParams;
+import io.katharsis.repository.ResourceRepository;
+import io.katharsis.servlet.resource.model.Node;
+
+/**
+ * Created by nickmitchell on 1/5/17.
+ */
+public class NodeRepository implements ResourceRepository<Node, Long> {
```
Comment (chb0github) [inline katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/NodeRepository.java]: see above

File: katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/LocaleRepository.java (hunk 3 above)
Comment (mitch2na) [inline katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/LocaleRepository.java]: I created this Locale resource to demonstrate that when you have an include relationship one level deep, it's fields marked with `@JsonApiIncludeByDefault` are ignored.

File: katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/NodeCommentRepository.java (hunk 4 above)
Comment (mitch2na) [inline katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/NodeCommentRepository.java]: Done

File: katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/NodeRepository.java (hunk 5 above)
Comment (mitch2na) [inline katharsis-servlet/src/test/java/io/katharsis/servlet/resource/repository/NodeRepository.java]: Done

**Pull request merged**